python/lsst/pipe/base/_dataset_handle.py

```diff
@@ -135,11 +146,31 @@ class can be found.
             else:
                 returnStorageClass = storageClass
 
+        inMemoryDataset = self.inMemoryDataset
+
+        if self.copy:
+            # An optimization might be to defer copying until any components
+            # and parameters have been applied. This can be a problem since
+            # most component storage classes do not bother to define a
+            # storage class delegate and the default delegate uses deepcopy()
+            # which can fail if explicit support for deepcopy() is missing
+            # or pickle does not work (for example VisitInfo fails).
```

Comment:
Is this saying that if you put a VisitInfo in an inMemoryDatasetHandle and use copy=True it will raise an exception on get?

Reply:
Yes. Because you can't pickle a VisitInfo.

```python
>>> copy.deepcopy(VisitInfo())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/timj/work/lsstsw/miniconda/envs/lsst-scipipe-6.0.0/lib/python3.10/copy.py", line 161, in deepcopy
    rv = reductor(4)
TypeError: cannot pickle 'lsst.afw.image._imageLib.VisitInfo' object
```

It looks like `VisitInfo` is immutable so I can add a `__deepcopy__` for it that returns itself.

Reply:
My feeling is that you are using `InMemoryDatasetHandle` in new code and will (1) know if you need to copy it or not (2) write a new delegate copy subclass if you really do need to copy it some special way, or (3) add a `__deepcopy__` method

Reply:
I have added `VisitInfo.__deepcopy__` in lsst/afw#686